lib/compat/wordpress-5.9/block-template-utils.php

```diff
@@ -36,7 +36,11 @@ function get_block_theme_folders( $theme_stylesheet = null ) {
 		$root_dir   = get_theme_root( $theme_name );
 		$theme_dir  = "$root_dir/$theme_name";
 
-		if ( is_readable( $theme_dir . '/block-templates/index.html' ) ) {
+		$parent_theme_name = get_template();
+		$parent_root_dir   = get_theme_root( $parent_theme_name );
+		$parent_theme_dir  = "$parent_root_dir/$parent_theme_name";
+
+		if ( is_readable( $theme_dir . '/block-templates/' ) || is_readable( $parent_theme_dir . '/block-templates/' ) ) {
```

Comment:
Can't we just remove `|| is_readable( $parent_theme_dir . '/block-templates/' )`, since we're calling the function for both themes (parent and child), this should work right?

Reply:
I don't think so. If we did that then we'd be back to the changes that are already in trunk, since the other lines don't actually do anything except assign variables.

Reply:
I thought the issue was about themes that are child themes, override some templates but not "index.html" so it's different than trunk? Are there any other use-cases here?

Reply:
They might not override any templates at all.

Reply:
> They might not override any templates at all.

In that case that function is useless and doesn't matter what it returns.

Reply:
If the parent defines templates inside `block-templates` but the child does not then this function looks for template parts inside `parts` inside the parent, which doesn't exist.

Reply:
Could this be solved by changing the check to `is_readable( $theme_dir . '/block-templates/' ) || is_readable( $theme_dir . '/block-template-parts/' )` instead. I want to avoid having to deal with "parent/child" logic inside this function.

Reply:
I don't think so. Child themes can simply provide a different theme.json file and no other changes.

Reply:
> I don't think so. Child themes can simply provide a different theme.json file and no other changes.

In that case it doesn't matter what this function returns.

Reply:
If I create a child theme with no templates or template parts I see this message:

<img width="1440" alt="Screenshot 2021-12-13 at 11 08 48" src="https://user-images.githubusercontent.com/275961/145801911-47ffddfa-d174-4ef1-97da-826453532227.png">

Reply:
Indeed, I think we had an issue with the template part block's code where the child theme was not handled properly and it should be fixed in 004cba5fc79d8f1ec9d9ee8791775baa87d2f247

The root issue is that `get_theme_file_path` function tries to fallback to the parent theme automatically but it doesn't account with the fact that the path could change slightly between parent and child themes (folder names)

Reply:
Feel free to ship this if the fix in that commit is good enough for you.

Reply:
Looks good thanks